src/provider/APIHandler.ts

```diff
@@ -14,11 +16,26 @@ export type ActionHandlerMap<Actions extends string, Payloads extends {[K in Act
 
 export class APIHandler<A extends string, P extends {[K in A]: unknown}, R extends {[K in A]: unknown}> {
     private _providerChannel!: ChannelProvider;
+    private _connectionSignal: Signal1<Identity>;
+    private _disconnectionSignal: Signal1<Identity>;
```

Comment:
Can we use the layouts pattern for defining/naming signals?

e.g. `public readonly onConnection: Signal1<Identity> = new Signal1();`